### PR TITLE
Add javascript-node and typescript-node

### PR DESCRIPTION
**Pull request by chrmarti**

### CODE REVIEW

File: src/javascript-node/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+| Metadata | Value |  
+|----------|-------|
+| *Contributors* | The VS Code Team |
+| *Categories* | Core, Languages |
+| *Definition type* | Dockerfile |
+| *Published image* | mcr.microsoft.com/vscode/devcontainers/javascript-node |
+| *Available image variants* | 18 / 18-bullseye, 16 / 16-bullseye, 14 / 14-bullseye, 18-buster, 16-buster, 14-buster ([full list](https://mcr.microsoft.com/v2/vscode/devcontainers/javascript-node/tags/list)) |
+| *Published image architecture(s)* | x86-64, arm64/aarch64 for `bullseye` variants |
+| *Works in Codespaces* | Yes |
```
Comment (samruddhikhandale) [inline src/javascript-node/README.md:15]: Remove? 
As we don't want Codespaces and Remote Containers specific details in this new repo.

File: src/javascript-node/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+| *Container host OS support* | Linux, macOS, Windows |
+| *Container OS* | Debian |
+| *Languages, platforms* | Node.js, JavaScript |
+
+See **[history](history)** for information on the contents of published images.
+
+## Using this definition
```
Comment (samruddhikhandale) [inline src/javascript-node/README.md:22]:
```suggestion
## Using this image
```

File: src/javascript-node/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+While the definition itself works unmodified, you can select the version of Node.js the container uses by updating the `VARIANT` arg in the included `devcontainer.json` (and rebuilding if you've already created the container).
```
Comment (samruddhikhandale) [inline src/javascript-node/README.md:24]: Can we remove `definitions/templates` specific data from this README ?

File: src/javascript-node/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+```jsonc
+// Or you can use 16-bullseye or 16-buster if you want to pin to an OS version
+"args": { "VARIANT": "16" }
+```
+
+You can also directly reference pre-built versions of `.devcontainer/base.Dockerfile` by using the `image` property in `.devcontainer/devcontainer.json` or updating the `FROM` statement in your own `Dockerfile` with one of the following:
+
+- `mcr.microsoft.com/vscode/devcontainers/javascript-node` (latest)
```
Comment (samruddhikhandale) [inline src/javascript-node/README.md:33]: Can we replace all `mcr.microsoft.com/vscode/devcontainers/*` references with `mcr.microsoft.com/devcontainers/*` ?

File: src/javascript-node/manifest.json [hunk 5]
```diff
@@ -0,0 +1,51 @@
+{
+	"variants": ["18-bullseye", "16-bullseye", "14-bullseye", "18-buster", "16-buster", "14-buster"],
+	"definitionVersion": "0.204.2",
```
Comment (samruddhikhandale) [inline src/javascript-node/manifest.json:3]:
```suggestion
	"imageVersion": "0.204.2",
```

File: src/typescript-node/.devcontainer.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
+{
+	"name": "Node.js & TypeScript",
+	"build": {
+		"dockerfile": "Dockerfile",
+		// Update 'VARIANT' to pick a Node version: 18, 16, 14.
+		// Append -bullseye or -buster to pin to an OS version.
+		// Use -bullseye variants on local on arm64/Apple Silicon.
+		"args": {
```
Comment (samruddhikhandale) [inline src/typescript-node/.devcontainer.json:8]: Same comment as above, can we remove `args`?

File: src/typescript-node/README.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
+# Node.js & TypeScript
+
+## Summary
+
+*Develop Node.js based applications in TypeScript. Includes Node.js, eslint, nvm, yarn, and the TypeScript compiler.*
+
+| Metadata | Value |  
+|----------|-------|
+| *Contributors* | The VS Code Team |
```
Comment (samruddhikhandale) [inline src/typescript-node/README.md:9]: Same comments here like the javascript-node's README 😄

File: src/typescript-node/manifest.json [hunk 8]
```diff
@@ -0,0 +1,60 @@
+{
+	"variants": ["18-bullseye", "16-bullseye", "14-bullseye", "18-buster", "16-buster", "14-buster"],
+	"definitionVersion": "0.204.2",
```
Comment (samruddhikhandale) [inline src/typescript-node/manifest.json:3]:
```suggestion
	"imageVersion": "0.204.2",
```

`imageVersion` is what is now used by the vscdc tooling.

File: src/javascript-node/.devcontainer.json [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+		"dockerfile": "Dockerfile",
+		// Update 'VARIANT' to pick a Node version: 18, 16, 14.
+		// Append -bullseye or -buster to pin to an OS version.
+		// Use -bullseye variants on local arm64/Apple Silicon.
+		"args": { "VARIANT": "16-bullseye" }
+	},
+    "features": {
+        "ghcr.io/devcontainers/features/common-utils:1": {
+            "installZsh": "true",
+            "upgradePackages": "true"
```
[the remark this answers is not in the document]
Comment (chrmarti) [reply src/javascript-node/.devcontainer.json:13]: The common-utils scripts appears to scan for well-known usernames. It is probably more robust to set it here. Adding that, thanks.

**Comment by chrmarti** [comment]
@samruddhikhandale Thanks for the detailed feedback! I have addressed all your comments. Could you take another look?

**Review by samruddhikhandale — APPROVED** [review]
It looks good! ✨  Thank you! 👏  

Once we have published the `javascript-node` tags with new path, I'll update the reference `mcr.microsoft.com/vscode/devcontainers/javascript-node:0-${VARIANT}` in `src/typescript-node/Dockerfile`

**Pull request merged**